big-practice/src/js/pages/dashboard.page.js

```diff
@@ -42,22 +44,35 @@ class Dashboard {
     DocumentHelper.cleanErrorMessage(this.nationInput);
   }
 
-  // /**
-  //  * Modal form update will appear when clicking the update button in the table
-  //  *
-  //  * @param {element} item - Table update button
-  //  */
-  // tableUpdateBtn(item) {
-  //   const movieId = item.dataset.id;
+  /**
+   * Modal form update will appear when clicking the update button in the table
+   *
+   * @param {element} item - Table update button
+   */
+  async handleTableUpdate(item) {
+    const movieId = item.dataset.id;
+    const movieData = await apiService.get(`${MOVIES_API}/${movieId}`);
+
+    ModalHelper.showModal(this.modalForm);
+    this.nameMovieInput.value = movieData.name;
+    this.directorInput.value = movieData.director;
+    this.nationInput.value = movieData.nation;
+    this.form.setAttribute('data-id', movieId);
+  }
 
-  //   showModal(modalForm);
-  //   getApi(`${MOVIES_API}/${movieId}`, (movieData) => {
-  //     nameMovieInput.value = movieData.name;
-  //     directorInput.value = movieData.director;
-  //     nationInput.value = movieData.nation;
-  //     form.setAttribute('data-id', movieId);
-  //   });
-  // }
+  /**
+   * Query to all update buttons in the table
+   */
+  queryUpdateButtons() {
```

Comment:
Wrong purpose, Query means you only query an element, but you also add event and handle

Reply:
Fixed